### ISSUE TITLE
"/" offers to create the 1st subfolder...

### ISSUE THREAD

**Issue by fredstluka** [RealRaven2000/QuickFolders#40]
In the quickMove box, "/" offers to create a new subfolder.  Nice!
It does so even if the parent folder has no other subfolders yet.

Personally, I LIKE that, don't want you to change anything, but it
contradicts what Axel said in a 6/11/2020 3:06am email.  He said
it was intended to work only if there were already other subfolders.

> did you already discover the use of / (forward slash) in the 
> quickMove box?
> It is quite handy when you have forgotte whether you already 
> have a folder for customer "Fred". just type:
> cus/Fre
> and you will get matches. but also a possibility to create a new 
> folder (**only if the parent already has child folders**; then QF 
> assumes you may want to stick in multiple ones into the tree, 
> e.g. Customers/Axel, Customers/Bill, Customers/Fred)

QF version 4.18.1
TB version 68.9.0
Mac OS X version 10.11.6
--Fred

**Comment by RealRaven2000** [comment]
> In the quickMove box, "/" offers to create a new subfolder. Nice!
> It does so even if the parent folder has no other subfolders yet.
> 
> Personally, I LIKE that, don't want you to change anything, but it
> contradicts what Axel said in a 6/11/2020 3:06am email. He said
> it was intended to work only if there were already other subfolders.
> 

Ok that is when you start with the "/" and that would be a very clear command to QF that you want a child folder *from the current location*. (I had actually forgotten about this feature).

My typical use is like this
- "I want to drop something into customers / fred"
- "but I don't know whether I already have a fred subfolder?"

so I type cu/fr and get 2 possible outcomes:
- Fred exists and shows up on the top of the list (I press enter to move the mail)
- No matching subfolder (starting with fr) is found - this causes the menu item "creat subfolder fr..." to appear on top.

The reason that in this case you need siblings is that otherwise I would have the "create new subfolder" suggestion way too often which would be felt as clutter. Once you make a subfolder to any parent, QF assumes "you may probably want to add more".

So (in a nutshell) when you start with typing / this is a special case and means "start from current folder". btw, these match algorithm / logic is very hard to program. I would be super careful before I change any behavior as touching this code is "regression heaven".

**Comment by fredstluka** [comment]
> Ok that is when you start with the "/" 

No.&nbsp;  I'm typing cu/fr like you describe below.&nbsp;  I'm NOT starting with a / as the 1st char in the quickMove box.

> and that would be a very clear command to QF that you want a child folder _from the current location_. (I had actually forgotten about this feature).

I'm not seeing that feature at all.&nbsp;  If I type /abcd in the quickMove box, it does nothing.&nbsp;  No dropdown.&nbsp;  No creation of subfolder.&nbsp;  Nothing at all.&nbsp; Even when it hit Enter.


>     * Fred exists and shows up on the top of the list (I press enter to move the mail)

Right.

>     * No matching subfolder (starting with fr) is found - this causes the menu item "creat subfolder fr..." to appear on top.

Right.&nbsp;  But this happens regardless of whether customers has any subfolders.&nbsp;  That's fine with me, but it doesn't match the docs.

> otherwise I would have the "create new subfolder" suggestion way too often which would be felt as clutter.

No.&nbsp;  For me, it always appears.&nbsp;  And that's fine.&nbsp;  Doesn't seem like clutter to me.

> btw, these match algorithm / logic is very hard to program. 

OK.&nbsp;  Don't change it.&nbsp;  I like it exactly the way it is.&nbsp;  Just confirm on your machine that it always offers to create a subfolder, even if it's the 1st subfolder.&nbsp; So we know it's consistent across Windows and Mac.&nbsp;  Otherwise, there's some bug here we haven't yet identified.&nbsp;  If it works the same for you as for me, just update the docs.

> I would be super careful before I change any behavior as touching this code is "regression heaven".

Yeah, in the Agile S/W Development world, we call that "brittle" code because no one wants to touch it for fear of breaking something.&nbsp;  I've seen that on lots of projects that I joined late.&nbsp;  The recommended solution is to add enough automated regression tests that you can "refactor fearlessly".

No need to go back now and write tests for everything, but perhaps when you hit a brittle case like this and DO want to make a risky change:
1. Set up an automated regression test mechanism
2. Write a few tests in the specific area that's likely to break if you fix the bug
3. Write a test case that demonstrates the bug
4. Run the automated test suite to see it fail because the bug occurred
5. Fix the bug
6. Run the automated test suite to see it no longer fail and that all the related tests still pass as well.

Over time, you'll gradually accumulate a regression test suite and will be able to refactor more and more fearlessly.

--Fred
